frontend/src/components/Pages/InspectionPage/InspectionSection.tsx

```diff
@@ -114,82 +170,137 @@ export function InspectionSection({ refreshInterval }: RefreshProps) {
-        for (const areaId of Object.keys(areaMissions)) {
-            for (const inspection of areaMissions[areaId].inspections) {
-                const missionDefinition = inspection.missionDefinition
-                const missionRuns = await BackendAPICaller.getMissionRuns({
-                    statuses: [MissionStatus.Paused, MissionStatus.Pending, MissionStatus.Ongoing],
-                    missionId: missionDefinition.id,
-                })
-                newOngoingMissions[missionDefinition.id] = missionRuns.content.length > 0
-            }
-        }
-        setOngoingMissions(newOngoingMissions)
-    }
-
     useEffect(() => {
         const id = setInterval(() => {
             updateOngoingMissionsMap(deckMissions)
         }, refreshInterval)
         return () => clearInterval(id)
     }, [deckMissions])
 
+    const getDeadlineInspection = (deadline: Date) => {
+        const deadlineDays = getDeadlineInDays(deadline)
+        switch (true) {
+            case deadlineDays <= 1:
+                return 'red'
+            case deadlineDays > 1 && deadlineDays <= 14:
+                return 'orange'
+            case deadlineDays > 7:
+                return 'green'
+        }
+    }
+
     const getCardColor = (deckId: string) => {
         const inspections = deckMissions[deckId].inspections
         if (inspections.length === 0) return 'gray'
         const isScheduled = (m: CondensedMissionDefinition) =>
             Object.keys(ongoingMissions).includes(m.id) && ongoingMissions[m.id]
         const sortedInspections = inspections.filter((i) => !isScheduled(i.missionDefinition)).sort(compareInspections)
+        console.log(sortedInspections)
         if (sortedInspections.length === 0) return 'green'
 
         const nextInspection = sortedInspections[0]
-        if (!nextInspection.deadline) return 'green'
 
-        const deadlineDays = getDeadlineInDays(nextInspection.deadline)
-        switch (true) {
-            case deadlineDays <= 1:
-                return 'red'
-            case deadlineDays > 1 && deadlineDays <= 14:
-                return 'yellow'
-            case deadlineDays > 7:
-                return 'green'
+        if (!nextInspection.deadline) {
+            if (!nextInspection.missionDefinition.inspectionFrequency) return 'gray'
+            else return 'red'
+        }
+
+        return getDeadlineInspection(nextInspection.deadline)
+    }
+
+    function getCardMissionInformation(deckId: string) {
+        var colorsCount: DeckMissionCount = {
+            red: { count: 0, message: 'Must be inspected this week' },
+            orange: { count: 0, message: 'Must be inspected within two weeks' },
+            green: { count: 0, message: 'Up to date' },
+            grey: { count: 0, message: '' },
         }
+        const inspections = deckMissions[deckId].inspections
+        if (inspections.length === 0) return
+
+        deckMissions[deckId].inspections.map((inspection) => {
```

Comment:
```suggestion
        deckMissions[deckId].inspections.foreach((inspection) => {
```
Get warning from using map when there is no return value

Reply:
![image](https://github.com/equinor/flotilla/assets/33518988/de038d4c-7a9d-4449-9cfb-5ced86e5ded0)

Reply:
foreach was not being supported for some reason so i added a return at the end to remove the error